sjsonnet/test/src/sjsonnet/UnicodeHandlingTests.scala

```diff
@@ -96,26 +109,55 @@ object UnicodeHandlingTests extends TestSuite {
       eval("std.sort(['\\uD800\\uDC00', '\\uFFFF'])") ==> ujson.Arr("\uFFFF", "\uD800\uDC00")
     }
 
-    // Unpaired surrogate handling - sjsonnet-specific behavior
-    //
-    // Note: This is an intentional divergence from go-jsonnet and C++ jsonnet:
-    // - go/C++ reject unpaired surrogates in escape sequences at parse time
-    // - go-jsonnet's std.char() replaces surrogate codepoints with U+FFFD
-    // - sjsonnet preserves unpaired surrogates throughout
-    //
-    // This permissive behavior is maintained for backwards compatibility.
+    // Updated surrogate handling tests - new parser correctly handles surrogate pairs
+    test("surrogateParHandling") {
+      // Valid surrogate pairs are now correctly parsed
+      eval("\"\\uD83D\\uDE00\"") ==> ujson.Str("😀") // Valid surrogate pair (grinning face)
+      eval("\"\\uD83C\\uDF0D\"") ==> ujson.Str("🌍") // Earth emoji
 
-    test("unpairedSurrogatesInEscapes") {
-      // sjsonnet parses these successfully (go/C++ would reject)
-      eval("\"\\uD800\"") ==> ujson.Str("\uD800") // High surrogate alone
-      eval("\"\\uDC00\"") ==> ujson.Str("\uDC00") // Low surrogate alone
-      eval("\"\\uD83D\\uDE00\"") ==> ujson.Str("😀") // Valid surrogate pair
+      // Test that the parser correctly converts surrogate pairs to proper Unicode characters
+      eval("std.codepoint(\"\\uD83C\\uDF0D\")") ==> ujson.Num(127757) // Earth emoji codepoint
     }
 
-    test("stdCharPreservesRawSurrogates") {
-      // sjsonnet preserves raw surrogate codepoints (go-jsonnet would replace with U+FFFD)
-      eval("std.codepoint(std.char(55296))") ==> ujson.Num(55296) // 0xD800 high surrogate
-      eval("std.codepoint(std.char(56320))") ==> ujson.Num(56320) // 0xDC00 low surrogate
```

Comment:
I think that we should continue to have these tests for `std.char()` behavior with raw surrogate codepoints, especially because the two other jsonnet implementations differ in how they handle these cases.

I think it's still worthwhile to document the behavior of these two cases in the new `stdCharHandling` test.

Reply:
Would you like commit directly in it, thanks

Reply:
I have added it back